packages/renderer/src/lib/dashboard/DashboardPage.svelte

```diff
@@ -1,18 +1,213 @@
+  if (dashboardOrdering.size === 0) {
+    return dashboardSections;
+  }
+  return [...dashboardSections].toSorted((a, b) => {
+    const aOrder = dashboardOrdering.get(a.id) ?? a.originalOrder;
+    const bOrder = dashboardOrdering.get(b.id) ?? b.originalOrder;
+    return aOrder - bOrder;
+  });
+}
+
+// Save configuration (like Table.svelte does)
+async function saveDashboardConfiguration(): Promise<void> {
+  if (tablePersistence.storage) {
+    const orderedItems = getOrderedDashboardSections();
+    const serializableItems = orderedItems.map(item => ({
+      id: item.id,
+      label: item.label,
+      enabled: item.enabled,
+      originalOrder: item.originalOrder,
+    }));
+
+    await tablePersistence.storage.save('dashboard', serializableItems);
+  }
+}
+
+// Initialize dashboard on mount
+onMount(async () => {
+  await initializeDashboard();
+});
+
+// Save configuration whenever dashboardSections or ordering changes
+$effect(() => {
+  if (isInitialized && dashboardSections.length > 0) {
+    dashboardPageRegistry.entries = updatedEntries;
+  }
+});
+
+// Reset function for dashboard layout
+async function resetDashboardLayout(): Promise<void> {
+  if (tablePersistence.storage) {
+    try {
+      // Reset using the persistence callbacks (clears saved config)
+      dashboardSections = await tablePersistence.storage.reset('dashboard', defaultSectionNames);
+      dashboardOrdering.clear();
+
+      // Reset the registry to default state
+      setupDashboardPageRegistry();
+
+      // Ensure the registry reflects the reset state
+      if (dashboardPageRegistry.entries.length > 0) {
+        const updatedEntries = convertFromListOrganizerItems(dashboardSections, dashboardPageRegistry.entries);
+        dashboardPageRegistry.entries = updatedEntries;
+      }
+    } catch (error: unknown) {
+      console.error('Failed to reset dashboard layout:', error);
+    }
+  }
+}
+
+// Handle dashboard order changes from ListOrganizer
+function handleDashboardOrderChange(newOrdering: SvelteMap<string, number>): void {
+  dashboardOrdering = new SvelteMap(newOrdering);
+
+  // Save configuration immediately when order changes
+  saveDashboardConfiguration().catch((error: unknown) => {
+    console.error('Failed to save dashboard configuration after order change:', error);
+  });
+}
+
+// Handle dashboard section toggle changes from LayoutEditor
+function handleDashboardToggle(itemId: string, enabled: boolean): void {
+  dashboardSections = dashboardSections.map(item => (item.id === itemId ? { ...item, enabled } : item));
+
+  // Save configuration immediately when toggle changes
+  saveDashboardConfiguration().catch((error: unknown) => {
+    console.error('Failed to save dashboard configuration after toggle change:', error);
+  });
+}
+
+// Filter and sort dashboard registry items based on LayoutEditor configuration
+let sortedDashboardRegistry = $derived.by(() => {
+  // Get ordered sections inline to ensure reactivity
```

Comment:
question: why is this needed to "Get ordered sections inline to ensure reactivity"?

I tried to remove this 

```ts

 const orderedSections =
    dashboardOrdering.size === 0
      ? [...dashboardSections]
      : [...dashboardSections].toSorted((a, b) => {
          const aOrder = dashboardOrdering.get(a.id) ?? a.originalOrder;
          const bOrder = dashboardOrdering.get(b.id) ?? b.originalOrder;
          return aOrder - bOrder;
        });

```

and it seems to be reactive when testing manually: I can still hide, show, move and reset layout.